### PR TITLE
pkg/payload/image: avoid rendering manifests

### PR DESCRIPTION
**Pull request by markmc**
The `image` command currently loads the image-references file and then
renders all manifests. Clearly the manifest rendering can be avoided
when we just want to look up an image reference.

This command is currently run 17 times by bootkube, and my quick test
shows this refactor halving the roughly 10 seconds this takes.

The main refactoring here is to split the loadUpdatePayloadMetadata()
function into getPayloadTasks() and loadImageReferences() so that the
latter can be used standalone in this particular case.

### CODE REVIEW

File: pkg/payload/image.go [hunk 1]
```diff
@@ -4,17 +4,23 @@ import (
...
 
 	"github.com/pkg/errors"
+	"k8s.io/klog"
 )
 
 // ImageForShortName returns the image using the updatepayload embedded in
 // the Operator.
 func ImageForShortName(name string) (string, error) {
-	up, err := LoadUpdate(DefaultPayloadDir, "")
+	klog.V(4).Infof("Loading image references from %q", DefaultPayloadDir)
```
Comment (abhinavdahiya) [inline pkg/payload/image.go]: why the logs?

File: pkg/payload/payload.go [hunk 2]
```diff
@@ -103,14 +103,23 @@ type Update struct {
 }
 
 func LoadUpdate(dir, releaseImage string) (*Update, error) {
-	payload, tasks, err := loadUpdatePayloadMetadata(dir, releaseImage)
+	klog.V(4).Infof("Loading updatepayload from %q", dir)
```
Comment (abhinavdahiya) [inline pkg/payload/payload.go]: why the logs?

File: pkg/payload/payload.go [hunk 3]
```diff
@@ -206,39 +215,34 @@ type payloadTasks struct {
...
-		return nil, nil, err
-	}
+func getPayloadTasks(dir, releaseImage string) []payloadTasks {
+	mrc := manifestRenderConfig{ReleaseImage: releaseImage}
 
-	imageRef, err := resourceread.ReadImageStreamV1(imageRefData)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "invalid image-references data %s", irf)
-	}
+	cjf := filepath.Join(dir, ReleaseManifestDir, cincinnatiJSONFile)
```
Comment (abhinavdahiya) [inline pkg/payload/payload.go]: keep the cvoDir and releaseDir.

File: pkg/payload/payload.go [hunk 4]
```diff
@@ -206,39 +215,34 @@ type payloadTasks struct {
...
-	// XXX: load cincinnatiJSONFile
-	cjf := filepath.Join(releaseDir, cincinnatiJSONFile)
-	// XXX: load imageReferencesFile
-	irf := filepath.Join(releaseDir, imageReferencesFile)
-	imageRefData, err := ioutil.ReadFile(irf)
-	if err != nil {
-		return nil, nil, err
-	}
+func getPayloadTasks(dir, releaseImage string) []payloadTasks {
+	mrc := manifestRenderConfig{ReleaseImage: releaseImage}
```
Comment (abhinavdahiya) [inline pkg/payload/payload.go]: move it to closer to where it's used.

File: pkg/payload/payload.go [hunk 5]
```diff
@@ -103,14 +103,23 @@ type Update struct {
 }
 
 func LoadUpdate(dir, releaseImage string) (*Update, error) {
-	payload, tasks, err := loadUpdatePayloadMetadata(dir, releaseImage)
+	klog.V(4).Infof("Loading updatepayload from %q", dir)
+	if err := ValidateDirectory(dir); err != nil {
```
Comment (abhinavdahiya) [inline pkg/payload/payload.go]: is there a case where we want to load from unvalidated directory?? If not, why make each caller to load* validate before calling them..?

File: pkg/payload/payload.go [hunk 6]
```diff
@@ -103,14 +103,23 @@ type Update struct {
 }
 
 func LoadUpdate(dir, releaseImage string) (*Update, error) {
```
Comment (abhinavdahiya) [inline pkg/payload/payload.go]: this function did 1) load metadata 2) start using it.
not this function does 1) validate dir, 2) load image reference 3) create an payload object 4) load tasks 

I think this not the ideal direction.. feel free to split the image referece and tasks, but don't remove the `loadUpdatePayloadMetadata` ??

File: pkg/payload/image.go (hunk 1 above)
Comment (markmc) [reply pkg/payload/image.go]: removed

File: pkg/payload/payload.go (hunk 6 above)
Comment (markmc) [reply pkg/payload/payload.go]: ok

**Pull request merged**